### ISSUE TITLE
Error when processing ValidationResults with ValidationResult.Success value

### ISSUE THREAD

**Issue by tjasek225** [luboshl/MiniValidationPlus#17]
When ValidationResults contains `ValidationResult.Success` which value is `null` `NullReferenceException` is thrown when processing validation results. In my test case it fails on the code line below, but it should be checked/filtered in all places where ValidationResult's properties are accessed.

https://github.com/luboshl/MiniValidationPlus/blob/a8a79c3b0a4629d1085904f0db687a228d132732/src/MiniValidationPlus/MiniValidatorPlus.cs#L706

**Comment by luboshl** [comment]
@tjasek225, thank you reporting another issue :) I'll investigate that and let you know.

**Comment by luboshl** [comment]
Documentation of `System.ComponentModel.DataAnnotations.ValidationResult` says:

```csharp
/// <summary>
///     Gets a <see cref="ValidationResult" /> that indicates Success.
/// </summary>
/// <remarks>
///     The <c>null</c> value is used to indicate success.  Consumers of <see cref="ValidationResult" />s
///     should compare the values to <see cref="ValidationResult.Success" /> rather than checking for null.
/// </remarks>
public static readonly ValidationResult? Success;
```

It's a strange design and it also gets nullablility warning in `IValidatableObject.Validate` method because it expects `IEnumerable<ValidationResult>`, not `IEnumerable<ValidationResult?>`. But I understant that this is for backward compatibility.

But the code in in MiniValidatorPlus should not fail bacause of that. You are right, the problem is when processing collection of `ValidationResult` in the `foreach` you mentioned.

I've created repro with failing test here: https://github.com/luboshl/MiniValidationPlus/commit/747743e8c07c7d125ac79dbb72f5619a4e61b01f

**Comment by luboshl** [comment]
@tjasek225, bug was fixed in #19 and released in [v1.2.2](https://github.com/luboshl/MiniValidationPlus/releases/tag/v1.2.2).
